packages/ses/src/commons.js

```diff
@@ -174,6 +176,89 @@ export const arraySome = uncurryThis(arrayPrototype.some);
+          // browsers, and on many other JS platforms.
+          // In those cases, we first use `structuredClone` to get a fresh
+          // buffer with exclusive access to the underlying data, while
+          // detaching it from the original `arrayBuffer`.
+
+          newLength = +newLength;
+          bigIntAsUintN(newLength, 0n);
+
+          const newBuffer = /** @type {ArrayBuffer} */ (
+            structuredClone(arrayBuffer, {
+              transfer: [arrayBuffer],
+            })
+          );
+          if (newLength >= newBuffer.byteLength) {
+            return newBuffer;
+          }
+          // If the requested length is shorter than the length of `buffer`,
+          // we use `slice` to shorted the returned result, but at the cost
+          // of an extra copy.
+          //
+          // `slice` accepts negative arguments but `transferToFixedLength`
+          // does not...
+          // get at the underlying ToIndex operation through `BigInt.asUintN`
+          // (avoiding the redundant allocation of e.g. `ArrayBuffer(newLength)`)
+          // and ToNumber through unary `+` (rather than `Number(newLength)`,
+          // which fails to reject BigInts).
+          //
+          // On platforms like Node 20
+          // - without`tranferToFixedLength` or `transfer`
+          // - with `structuredClone`
+          // - with `resize`
+          //
+          // It might seem like we could avoid the extra copy by
+          // `newBuffer.resize(newLength)`. But `structuredClone`
+          // makes ArrayBuffers that are not resizable.
+          return arrayBufferSlice(newBuffer, 0, newLength);
+        }
+      : /**
+         * @param {ArrayBuffer} arrayBuffer
+         * @param {number} newLength
+         * @returns {ArrayBuffer}
+         */
+        (arrayBuffer, newLength = arrayBuffer.byteLength) => {
+          // There is no `transferToFixedLength` on Node <= 20,
+          // and no `structuredClone` on Node <= 17 and possibly on some
+          // non-browser JavaScript platforms.
+          // In those cases,
+          // and assuming the absence of `transfer`, we cannot detach
+          // the original, but we must still produce a new fresh buffer with
+          // exclusive mutability of its underlying state. We use `slice`
+          // both to make this exclusive copy and size it appropriately.
+
+          // `slice` accepts negative arguments but `transferToFixedLength`
+          // does not...
+          // get at the underlying ToIndex operation through `BigInt.asUintN`
+          // (avoiding the redundant allocation of e.g. `ArrayBuffer(newLength)`)
+          // and ToNumber through unary `+` (rather than `Number(newLength)`,
+          // which fails to reject BigInts).
+          newLength = +newLength;
+          bigIntAsUintN(newLength, 0n);
+
+          const newBuffer = arrayBufferSlice(arrayBuffer, 0, newLength);
+          return newBuffer;
+        };
```

Comment:
```suggestion
        (arrayBuffer, newLength = arrayBuffer.byteLength) => {
          // There is no `transferToFixedLength` on Node <= 20,
          // and no `structuredClone` on Node <= 17 and possibly on some
          // non-browser JavaScript platforms.
          // In those cases,
          // and assuming the absence of `transfer`, we cannot detach
          // the original, but we must still produce a new fresh buffer with
          // exclusive mutability of its underlying state.

          // Before looking at actual arguments, verify that the input is an
          // ArrayBuffer.
          const srcLength = arrayBufferByteLength(arrayBuffer);

          // Calculate ToIndex(newLengthAsNumber) using `BigInt.asUintN`,
          // first getting newLengthAsNumber as ToNumber(newLength) using unary `+`
          // (rather than `Number(newLength)`, which fails to reject BigInts).
          newLength = +newLength;
          bigIntAsUintN(newLength, 0n);

          // We might be able to copy some or all of the contents.
          if (newLength <= srcLength) {
            const copied = arrayBufferSlice(arrayBuffer, 0, newLength);
            const copiedLength = arrayBufferByteLength(copied);
            if (copiedLength !== newLength) {
              throw RangeError(`internal: length ${copiedLength} should have been ${newLength}`);
            }
            return copied;
          }

          // We need a bigger boat.
          // NOTE: Uint8Array must be extracted from globalThis.
          const view = new Uint8Array(newLength);
          typedArraySet(view, new Uint8Array(arrayBuffer));
          return typedArrayBuffer(view);
        };
```